lesson3/hometask03_ankarut.rb

```diff
@@ -0,0 +1,54 @@
+require 'rspec'
+require 'rspec/expectations'
+include RSpec::Matchers
+
+class Array
+
+  def custom_each(obj_class=nil, &block)
+    i = 0
+    while i < self.length
+      el = self[i]
+      i+=1
+      yield(el) if block_given?
+    end
+    if obj_class
+      obj_class[self]
+    else
+      self
+    end
+  end
+end
+
+class Hash
+  def custom_each(&block)
+    if block_given?
+      Array.new(self.to_a).custom_each(self.class, &block)
```

Comment:
It doesn't look like really elegant way to share this piece of code.
Why don't you use `Module`?